### PR TITLE
add CI action for Cicero

### PR DESCRIPTION
**Pull request by dermetfan**
This adds a Cicero action that runs all hydra jobs on pushes and pull requests.
Also purified nix evaluation of hydra jobs.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -159,8 +159,8 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-node
-  tag: 43393ac88faa99986ecaa0be409e8b6353f5e9fe
-  --sha256: 0g2m1gj0jqsk8gwx19n1h6kh3xjywwvxzw5caqj4jwz5m2z1cmml
+  tag: 497e05522a5d909d671db4427f721f7d888ee4b4
+  --sha256: 1jdz2xpva232vi62662za5zvgs3y6p50fvgj17iqpn0dvv9n1dwb
```
Comment (dermetfan) [inline cabal.project:163]: see https://github.com/input-output-hk/cardano-node/pull/4518

Comment (dermetfan) [reply cabal.project:163]: This is following the `backport-purity-1.35.0-rc4` branch. In the future we can follow master again as the PR was merged.

**Comment by kderme** [comment]
@dermetfan is this finalised? Does it require a rebase or a review from @input-output-hk/devops

File: cabal.project (hunk 1 above)
Comment (dermetfan) [reply cabal.project]: Removed as the node PR was merged so this is no longer needed.

**Comment by dermetfan** [comment]
Cicero reports this error for `nix build .#ciJobs.x86_64-linux.required` which is equivalent to `nix build required -f release.nix` on current master:

```
error: A definition for option `services.cardano-db-sync.socketPath' is not of type `null or (path)'. Definition values:
       - In `/nix/store/xxs457ajxj8fjm2shjmik6mdz2spyjzy-source/nixos/lib/build-vms.nix': <function>
```

As it fails with the same message on master as well I would say CI is doing its job and this is ready to merge. @kderme

**Review by jbgi — APPROVED** [review]

**Review by kderme — APPROVED** [review]
re cabal.project changes

**Review by jbgi — APPROVED** [review]

**Review by olgahryniuk — APPROVED** [review]

**Review by CarlosLopezDeLara — APPROVED** [review]

**Pull request merged**